app/components/chain-selector.tsx

```diff
@@ -42,6 +45,34 @@ export function ChainSelector() {
   const [open, setOpen] = React.useState(false)
   const currentChainId = useCurrentChain()
   const setChainId = useServerStore((s) => s.setChainId)
+  const { switchChain } = useSwitchChain()
+  const baseMint = useServerStore((s) => s.baseMint)
+  const router = useRouter()
+
+  /**
+   * Handles chain selection with dual strategy:
+   * - Multichain tokens: Switch chain directly (no navigation needed)
+   * - Single-chain tokens: Navigate to WETH with chain parameter to avoid
+   *   mismatch modals between token chain and app chain state
+   */
+  const handleChainSelect = (chainValue: string) => {
+    const chainId = Number.parseInt(chainValue) as ChainId
+    const isMultiChain = isAddressMultiChain(baseMint)
+
+    if (!isMultiChain) {
+      // Navigate to WETH (universal fallback) with chain parameter
+      // Page will read 'c' param and set chain state after URL-based token state
+      router.push(`/trade/WETH?c=${chainId}`)
```

Comment:
Why do we do this pattern of temporarily writing the chainId to the url? In `page-client`, all we do is `setChainId` and `switchChain` anyway, is there no way to just do that here but _also_ set the traded token to WETH?

This just feels like a hacky pattern that will be easy to forget about, and of course I want to be sure that the extra redirect won't impact UX